test/integration/concierge_tls_spec_test.go

```diff
@@ -0,0 +1,221 @@
+			kind: WebhookAuthenticator
+			metadata:
+				name: invalid-webhook-auth-missing-key
+			spec:
+				endpoint: "https://web-auth/token"
+				tls:
+					certificateAuthorityDataSource:
+						kind: Secret
+						name: foo
+			`)),
+			customResourceName: "invalid-webhook-auth-missing-key",
+			expectedError:      `The WebhookAuthenticator "invalid-webhook-auth-missing-key" is invalid: spec.tls.certificateAuthorityDataSource.key: Required value`,
+		},
+		{
+			name: "should disallow certificate authority data source with empty value for key",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: authentication.concierge.pinniped.dev/v1alpha1
+			kind: WebhookAuthenticator
+			metadata:
+				name: invalid-webhook-auth-empty-kind
+			spec:
+				endpoint: "https://web-auth/token"
+				tls:
+					certificateAuthorityDataSource:
+						kind: Secret
+						name: foo
+						key: ""
+			`)),
+			customResourceName: "invalid-webhook-auth-empty-kind",
+			expectedError:      `The WebhookAuthenticator "invalid-webhook-auth-empty-kind" is invalid: spec.tls.certificateAuthorityDataSource.key: Invalid value: "": spec.tls.certificateAuthorityDataSource.key in body should be at least 1 chars long`,
+		},
+		{
+			name: "should disallow certificate authority data source with missing kind",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: authentication.concierge.pinniped.dev/v1alpha1
+			kind: WebhookAuthenticator
+			metadata:
+				name: invalid-webhook-auth-missing-kind
+			spec:
+				endpoint: "https://web-auth/token"
+				tls:
+					certificateAuthorityDataSource:
+						name: foo
+						key: bar
+			`)),
+			customResourceName: "invalid-webhook-auth-missing-kind",
+			expectedError:      `The WebhookAuthenticator "invalid-webhook-auth-missing-kind" is invalid: spec.tls.certificateAuthorityDataSource.kind: Required value`,
+		},
+		{
+			name: "should disallow certificate authority data source with empty value for kind",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: authentication.concierge.pinniped.dev/v1alpha1
+			kind: WebhookAuthenticator
+			metadata:
+				name: invalid-webhook-auth-invalid-kind
+			spec:
+				endpoint: "https://web-auth/token"
+				tls:
+					certificateAuthorityDataSource:
+						kind: ""
+						name: foo
+						key: bar
+			`)),
+			customResourceName: "invalid-webhook-auth-invalid-kind",
+			expectedError:      `The WebhookAuthenticator "invalid-webhook-auth-invalid-kind" is invalid: spec.tls.certificateAuthorityDataSource.kind: Unsupported value: "": supported values: "Secret", "ConfigMap"`,
+		},
+		{
+			name: "should disallow certificate authority data source with invalid kind",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: authentication.concierge.pinniped.dev/v1alpha1
+			kind: WebhookAuthenticator
+			metadata:
+				name: invalid-webhook-auth-invalid-kind
+			spec:
+				endpoint: "https://web-auth/token"
```

Comment:
The controller will really call this endpoint, so better to use the real test web hook from the integration test environment.

Reply:
This test validates the kubebuilder validations that are part of the spec. So these resources can be created in the `default` namespace and don't have to be created in the `pinniped-supervisor` namespace. Because the resources are created in the `default` namespace, they will not be reconciled by the controllers. So just in this test, it is OK to use dummy data without having to add complexity to generating the test data. I am going to mark this as resolved. Please feel free to unresolve.